### PR TITLE
chore(knobs_example): migrate knobs example to widgetbook-3

### PR DESCRIPTION
**Pull request by Roaa94**
Migrates knobs example to widgetbook-3

### CODE REVIEW

File: examples/knobs_example/pubspec.lock [hunk 1]
```diff
@@ -223,14 +223,14 @@ packages:
       path: "../../packages/widgetbook"
       relative: true
     source: path
-    version: "3.0.0-beta.3"
+    version: "3.0.0-beta.4"
```
Comment (jenshor) [inline examples/knobs_example/pubspec.lock:226]: version `3.0.0-beta.4` has a navigation problem for the properties. https://github.com/widgetbook/widgetbook/pull/418 fixes that but I already deployed `3.0.0-beta.5` so you might want to run `melos bs` again

**Review by jenshor — APPROVED** [review]
LGTM 🎉

File: examples/knobs_example/pubspec.lock (hunk 1 above)
Comment (Roaa94) [reply examples/knobs_example/pubspec.lock]: I ran `melos bs` and pushed again but I'm not sure if it's right to include `pubspec` changes of other packages in this PR. Let me know if that's an issue and I can revert that.

**Review by jenshor — APPROVED** [review]
LGTM 🎉

**Pull request merged**